lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -398,6 +398,8 @@ $(document).ready(function() {
     $('span.critic.comment').filter(function() {return $(this).text()!="";}).before('&#8225;');
   }
 
+  $('.markdown-body p, .markdown-body span').attr('dir','auto');
```

Comment:
What's the reason for doing this with JS and not in de template?

Reply:
I don't mind it, just wondering.

Reply:
These `p` and `span` tags are the one generated by the rendering gem, so they're not there yet in the template.

Reply:
👍

Reply:
I think `p` and `span` covers it, or do you know of other block-level elements generated by the markup?

Reply:
Right, `pre` should probably be on the list, as should `table`. 👍

If I understand correctly, `dir` is a global attribute that *can* be added to any element, including `ul`, `h`, etc. But adding it to every tag seems to be [not recommended](https://www.w3.org/International/questions/qa-html-dir#markkupcrazy). The question is what's reasonable. Is it it enough to give the user the option of *changing* direction on the level of `p`, `span`, etc., or do we want to add the tags you meant?

Reply:
I should add that I think it is ultimately the responsibility of markup specifications to support RTL in a sane way. There's some talk of adding RTL indication to markdown, but I don't think it exists yet: https://talk.commonmark.org/t/explicit-rtl-indication-in-pure-markdown/286
Again, the question is what's reasonable for us to do when it comes to switching direction (if you use only an RTL script on the page, that should work fine with the current implementation)

Reply:
>  adding it to every tag seems to be not recommended.

That makes sense! My intuition says that users or RTL would want to have their entire page, including the headers, to be RTL. Changing direction mid-page and only for certain elements seems niche-behavior to me. Just adding `pre` and `table` should be fine. 👍